app/controllers/api/alert_definitions_controller.rb

```diff
@@ -1,26 +1,31 @@
 module Api
   class AlertDefinitionsController < BaseController
-    REQUIRED_FIELDS = %w(description db expression options).freeze
 
     before_action :set_additional_attributes
 
     def create_resource(type, id, data = {})
       assert_id_not_specified(data, type)
-      assert_all_required_fields_exists(data, type, REQUIRED_FIELDS)
       begin
-        data["expression"] = MiqExpression.new(data["expression"])
+        create_miq_expression(data)
         data["enabled"] = true if data["enabled"].nil?
-        super(type, id, data).serializable_hash.merge("expression" => data["expression"])
+        super(type, id, data.deep_symbolize_keys).serializable_hash.merge("expression" => expression_values(data).first)
```

Comment:
Why is the `deep_symbolize_keys` needed? If it's for a particular serialized column that is keyed with symbols, I think it would be better to make that explicit rather than converting everything wholesale

Reply:
We want the created alerts via API to be identical to alerts created in other places. 
It is not a particular (single) column, so I think it will be nicer not to list them.